subsys/bluetooth/host/smp.c

```diff
@@ -4369,8 +4369,9 @@ static int bt_smp_recv(struct bt_l2cap_chan *chan, struct net_buf *buf)
 
 	if (!atomic_test_and_clear_bit(smp->allowed_cmds, hdr->code)) {
 		LOG_WRN("Unexpected SMP code 0x%02x", hdr->code);
-		/* Don't send error responses to error PDUs */
-		if (hdr->code != BT_SMP_CMD_PAIRING_FAIL) {
+		/* Don't send error responses outside of pairing procedure and to error PDUs */
+		if (atomic_test_bit(smp->flags, SMP_FLAG_PAIRING) &&
+		    hdr->code != BT_SMP_CMD_PAIRING_FAIL) {
 			smp_error(smp, BT_SMP_ERR_UNSPECIFIED);
 		}
```

Comment:
I think this is easier to read in early-return style.

```suggestion
		if (!atomic_test_bit(smp->flags, SMP_FLAG_PAIRING)) {
			return 0;
		}
		if (hdr->code == BT_SMP_CMD_PAIRING_FAIL) {
			return 0;
		}
		smp_error(smp, BT_SMP_ERR_UNSPECIFIED);
```